realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -86,6 +86,37 @@
     // we register the listener.
     private final AtomicLong progressListenerId = new AtomicLong(-1);
 
+    private static final byte STATE_VALUE_WAITING_FOR_ACCESS_TOKEN = 0;
+    private static final byte STATE_VALUE_ACTIVE = 1;
+    private static final byte STATE_VALUE_DYING = 2;
+    private static final byte STATE_VALUE_INACTIVE = 3;
+    private static final byte STATE_VALUE_ERROR = 4;
+
+    public enum State {
+        STATE_WAITING_FOR_ACCESS_TOKEN(STATE_VALUE_WAITING_FOR_ACCESS_TOKEN),
+        STATE_ACTIVE(STATE_VALUE_ACTIVE),
+        STATE_DYING(STATE_VALUE_DYING),
+        STATE_INACTIVE(STATE_VALUE_INACTIVE),
+        STATE_ERROR(STATE_VALUE_ERROR);
+
+        final byte value;
+
+        State(byte value) {
+            this.value = value;
+        }
+
+        public static State fromByte(byte value) {
```

Comment:
This method is in the public API. Is that really necessary? If yes, it need Javadoc